### DISCUSSION TITLE
Loading assets/images dynamically from an array

### DISCUSSION THREAD

**Discussion by borsTiHD** [nuxt/nuxt#17693] [Questions]
I'm building my first nuxt 3 app and it's amazing.
However, I have come across a problem.

I want to declare an array with a set of images and loop with `v-for`.
Actually the data is supposed to come from a pinia store, but to simplify it, something along those lines below is what I'm trying to achieve.

I know in nuxt 2 I could use `require` and in nuxt 3 I should use `import`... but how could I `import` dynamically an unknown number of images from an array?

```js
<template>
    <div>
        <img
            v-for="(img, index) in images"
            :key="index"
            :src="`~/assets/img/${img}`"
            loading="lazy"
        >
    </div>
</template>

<script setup>
const images = [
    'image1.jpg',
    'image2.jpg'
    ...
]
</script>
```

I also found this issue: https://github.com/nuxt/framework/issues/2761
Unfortunately the answer with the reference to the Vite Docu does not help me.

Thank you.

**Comment by sewalsh** [comment]
https://modern-web.dev/docs/building/rollup-plugin-import-meta-assets/

Try something like this:

```js
<template>
    <div>
        <img
            v-for="(img, index) in images"
            :key="index"
            :src="images[index]"
            loading="lazy"
        >
    </div>
</template>

<script setup>
const images = [
    new URL("~/assets/img/image1.jpg", import.meta.url)
    new URL("~/assets/img/image2.jpg", import.meta.url)
    ...
]
</script>

```

**Reply by borsTiHD** [reply]
Thank you so much for the help. I tried it and was confronted by some errors. Then I realized that this is a plugin. Try to find out tomorrow how I can use this with nuxt 3. 😅

If anyone has any other ideas, preferably using onboard tools, I would be very grateful.

**Comment by borsTiHD** [comment]
If anyone else runs into this problem, these are my solutions to it so far.

**Solution 1:**
However, it is very useless to import 10 or more images this way. 🤓

```js
<script setup>
import img1 from '@/assets/img/img1.jpg'
import img2 from '@/assets/img/img1.jpg'
import img3 from '@/assets/img/img1.jpg'
...

const images = [
    img1,
    img2,
    img3,
    ...
]
</script>
```

**Solution 2:**
Another way is the "public" folder. This allows me to define the url the normal way in the array.
But I don't know if this is best practice, especially with bundling/ssr.
Can someone please explain to me what would be the preferable way?

```js
<script setup>
const images = [
    '/img/img1.jpg',
    '/img/img2.jpg',
    '/img/img3.jpg',
    ...
]
</script>
```

**Reply by benoitdemaegdt** [reply]
Hey 👋 

I am also using the public folder in my project (solution 2)

But I am also wondering if this is a good practice ... 🤔 
So far, I stick with this solution, because if I decide to host my images on something like cloudinary (or S3) I only need to change the URL of the image and it should still work.

In my opinion, solution 1 is too complicated. If your content is dynamic you will hardly be able to import all the images before using them.

Hopefully [nuxt/image should soon support nuxt3](https://github.com/nuxt/image/issues/500). I guess this module will provide best practices for using images in nuxt3 🤞

**Reply by benoitdemaegdt** [reply]
@clemcode  is writing some doc about it 👉  https://github.com/nuxt/framework/pull/4563/files
